cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -288,22 +288,46 @@ func (ng *AwsNodeGroup) AtomicIncreaseSize(delta int) error {
 // DecreaseTargetSize decreases the target size of the node group. This function
 // doesn't permit to delete any existing node and can be used only to reduce the
 // request for new nodes that have not been yet fulfilled. Delta should be negative.
-// It is assumed that cloud provider will not delete the existing nodes if the size
-// when there is an option to just decrease the target.
+// It is assumed that the cloud provider will not delete the existing nodes if there is an
+// option to just decrease the target.
 func (ng *AwsNodeGroup) DecreaseTargetSize(delta int) error {
+	// Ensure that the delta is negative to prevent an increase in size
 	if delta >= 0 {
-		return fmt.Errorf("size decrease size must be negative")
+		return fmt.Errorf("size decrease must be negative")
 	}
 
+	// Get the current size of the ASG
```

Comment:
Here and in what follows: the added comments I don't think provide much value.  Comments should explain _why_, not _what_, and saying things like "gets the current size of the ASG" are just repeating what the next line of code says.  My inclination here would be to remove most or all of the added comments to this change.